kube-insight-installation-in-GKE.md

```diff
@@ -0,0 +1,196 @@
+namespace "metrics" created
+```
+#### 3.1.3. Set user with cluster-admin for deploy RBAC
+```
+kubectl create clusterrolebinding cluster-admin-binding   --clusterrole cluster-admin   --user $(gcloud config get-value account) --cluster gke_test1012-219213_europe-north1-a_monitored
+```
+
+Note: if this step is missing, you will get following error during step 3.1.4
+```
+Error from server (Forbidden): error when creating "output/node-exporter/node-exporter-rbac.yaml": clusterroles.rbac.authorization.k8s.io "node-exporter" is forbidden: attempt to grant extra privileges: [PolicyRule{Resources:["tokenreviews"], APIGroups:["authentication.k8s.io"], Verbs:["create"]} PolicyRule{Resources:["subjectaccessreviews"], APIGroups:["authorization.k8s.io"], Verbs:["create"]}] user=&{li.wu@elastisys.com  [system:authenticated] map[user-assertion.cloud.google.com:[AGKDXmrmxR4ldy6MKeKjP/yU11f3oYRk8PD+N0PgGAq22xLez1St1GzcuML16US1LpSUuGhBOnxyxwZsHp7SCQjDpznRqruCdKWr/VWWNPbAwug79cP3/aWm082ChAydsV4O9mpVKPNRdUeQGWHrVs1nNYADq9+a39CgHjH6lHF0hGtRHFdZ+21irqPHQYoc3/FkERUC2ax9wRdypoU/lgYoRVj9o4siTnwHcmU=]]} ownerrules=[PolicyRule{Resources:["selfsubjectaccessreviews" "selfsubjectrulesreviews"], APIGroups:["authorization.k8s.io"], Verbs:["create"]} PolicyRule{NonResourceURLs:["/api" "/api/*" "/apis" "/apis/*" "/healthz" "/swagger-2.0.0.pb-v1" "/swagger.json" "/swaggerapi" "/swaggerapi/*" "/version"], Verbs:["get"]}] ruleResolutionErrors=[]
+```
+#### 3.1.4 Deployment
+`kubectl apply -R -f output/ --cluster gke_test1012-219213_europe-north1-a_monitored`
+
+After deployment, below services are available
+```
+manifestr)li_wu@cloudshell:~/kube-insight-manifests/manifests/agents/metrics (test1012-219213)$ kubectl get svc --cluster gke_test1012-219213_europe-north1-a_monitored -n metrics
+NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+kube-state-metrics    ClusterIP   10.35.248.48    <none>        8080/TCP         23m
+node-exporter         ClusterIP   None            <none>        9100/TCP         23m
+prometheus            ClusterIP   10.35.251.102   <none>        9090/TCP         23m
+prometheus-nodeport   NodePort    10.35.243.209   <none>        9090:30090/TCP   23m
+```
+### 3.2 Deploy logging
+#### 3.2.1 Render j2 files
+```
+(manifestr)li_wu@cloudshell:~/kube-insight-manifests/manifests/agents/logging (test1012-219213)$ manifestr --values `pwd`/values.py --template-root-dir `pwd`/templates --output-dir `pwd`/output
+```
+#### 3.2.2.  Create namespace logging
+`kubectl create ns logging  --cluster gke_test1012-219213_europe-north1-a_monitored`
+
+#### 3.2.3. Deployment
+```
+(manifestr)li_wu@cloudshell:~/kube-insight-manifests/manifests/agents/logging (test1012-219213)$ kubectl apply -R -f output/ --cluster gke_test1012-219213_europe-north1-a_monitored
+After deployment,  pods are:
```

Comment:
Maybe split this into two snippets?

Reply:
Thank you for your careful review, I just split it.